<launch>
	<node name="rviz" pkg="rviz" type="rviz" respawn="false" output="screen" args=""/>
	<node name="gps_test" pkg="sdu_pichi_surveyor" type="pichi_test.py" output="screen" args="$(find sdu_pichi_surveyor)/data/deviations.txt"/>
	<node name="rosbag_player" pkg="rosbag" type="play" args="$(find sdu_pichi_remote)/bags/sensors_only.bag" respawn="true"/>
	
	<node pkg="tf" type="static_transform_publisher" name="bottom_broadcaster" args="-1.23 0 0 0 0 0 imu_link mast_bottom 100"/>
	<node pkg="tf" type="static_transform_publisher" name="map_broadcaster" args="-588599.0922487933 -6137322.245287642 0 0 0 0 mast_top odom_combined 100"/>

	<!-- IMU -->
	<node pkg="vectornav_vn100" name="imu_node"  type="vectornav_vn100_node" output="screen" respawn="True" >
		<param name="vectornav_vn100_sub" value="/fmData/imu_rx"/>
		<param name="imu_pub" value="/fmInformation/imu"/>
		<param name="frame_id" value="mast_top"/>
		<param name="use_enu" value="true"/>
		<param name="covariance_x" value="0.001"/>
		<param name="covariance_y" value="0.001"/>
		<param name="covariance_z" value="0.001"/>
	 </node>

	<!--node pkg="serial_string" name="imu_serial_string_node" type="serial_string_node" output="screen" >
		<param name="publisher_topic" value="/fmData/imu_rx" />	 
		<param name="subscriber_topic" value="/fmData/imu_tx" />
		<param name="device" value="/dev/gps" />	
		<param name="baudrate" value="115200" />
		<param name="termination_character" value="10"/>
	</node--> 
	
	<!-- GPS -->
	<!--node pkg="serial_string" name="gps_serial" type="serial_string_node" output="screen" >
		<param name="publisher_topic" value="/fmData/gps_rx" />	 
		<param name="subscriber_topic" value="/fmData/gps_tx" />
		<param name="device" value="/dev/imu" />	
		<param name="baudrate" value="115200" />
		<param name="termination_character" value="10"/>
	</node-->
	
    <!--node pkg="serial_nmea" name="nmea_receiver" type="nmea_node" output="screen">
		<param name="str_to_msg_sub" value="/fmData/gps_rx" />   
		<param name="str_to_msg_pub" value="/fmData/nmea_from_gps" />
		<param name="msg_to_str_sub" value="/fmData/nmea_to_gps" />   
		<param name="msg_to_str_pub" value="/fmData/gps_tx" />           
		<param name="use_nmea_checksum" type="boolean" value="true" />
    </node-->	
    
    <node pkg="gps_gpgga" name="nmea_to_gpgga" type="gps_gpgga_node" output="screen">
        <param name="subscribe_topic_id" value="/fmData/gps_rx" />
		<param name="publish_topic_id" value="/fmInformation/gpgga" />
	</node>
	
    <node pkg="gpgga_to_utm" name="gpgga_to_utm_converter" type="gpgga_to_utm_node" output="screen"> 
        <param name="subscribe_topic_id" value="/fmInformation/gpgga" />
		<param name="publish_topic_id" value="/fmInformation/utm" />
	</node>
	
    <node pkg="utm_to_odom" name="gps_odometry" type="utm_to_odom_node" output="screen"> 
    	<param name="subscribe_topic_id" value="/fmInformation/utm" />
		<param name="publish_topic_id" value="/fmKnowledge/gps_odom" />
		<param name="odom_frame_id" value="gps_frame" />
		
		<param name="vehicle_frame" value="map" />
		<param name="odom_estimate_frame" value="odom_combined" />
		
		<param name="child_odom_frame_id" value="world_frame" />
		<param name="receive_n_before_publish" value="10"/>
		<param name="gps_variance" value="1"/>
	</node>

</launch>
